jetty-server/src/main/java/org/eclipse/jetty/server/ResponseWriter.java

```diff
@@ -456,11 +456,29 @@ public PrintWriter format(Locale l, String format, Object... args)
     { 
         try 
         {
+            
+            if(l==null) 
+                l = _locale;
+            
             synchronized (lock) 
             {
                 isOpen();
-                if ((_formatter == null) || (_formatter.locale() != l))
+
+                if(_formatter == null)
+                {
                     _formatter = new Formatter(this, l);
+                } 
+                else if(_formatter.locale() != null) 
+                {
+                    if(!_formatter.locale().equals(l))
+                        _formatter = new Formatter(this, l);
+                } 
+                else 
+                {
+                    if(l != null)
+                        _formatter = new Formatter(this, l);
+                }
+                
```

Comment:
I think this would read better as 
``` 
if (_formatter==null)
{ ... }
else if (_formatter.locale()==null)
{ ... }
else if (!_formatter.locale().equals(l))
{ ... }
```